core/src/workflows/application.rs

```diff
@@ -26,7 +35,18 @@ pub async fn initialize(
         context.log("dna/initialize: Initializing new chain from given DNA...");
         await!(initialize_chain(dna.clone(), &instance_context))?;
     }
+
+    // 3. Initialize the network
     await!(initialize_network(&instance_context))?;
+
+    // 4. Call publish on the agent and DNA entries. 
+    // This is to trigger the publishing of their headers not the entries themselves
+    let dna_entry = Entry::Dna(Box::new(dna.clone()));
```

Comment:
Isn't this going to get memory expensive because it's a full clone of the code?  Any way to stub that because the DNA entry is not actually used?

Reply:
It's not too bad as the WASM binary is hold in an `Arc` inside `DnaWasm` here: https://github.com/holochain/holochain-rust/blob/38837a4472387e2f14a312984ba5fa64e039b513/core_types/src/dna/wasm.rs#L96